### DISCUSSION TITLE
Ubuntu 22.04: Sage 9.8 and Apache Spark

### DISCUSSION THREAD

**Discussion by haraldschilly** [sagemathinc/cocalc#6544] [Announcements]
A new update of the upcoming Ubuntu 22.04 software environment has been released. Most notably:

* Sage 9.8 is the default now
* There is an Apache Spark instance in `/ext/apache/spark/current` and an associated `PySpark` kernel for Jupyter.

**Comment by williamstein** [comment]
> There is an Apache Spark instance in /ext/apache/spark/current and an associated PySpark kernel for Jupyter.

How can I try out some sort of "Hello World" with this?  I've never actually used Spark!

**Comment by haraldschilly** [comment]
Well, I would suggest to start understanding [RDDs](https://spark.apache.org/docs/latest/rdd-programming-guide.html). There are Python examples as well.

Here is a notebook: https://cocalc.com/haraldschilly/tests/pyspark

**Reply by williamstein** [reply]
Thanks, that worked well in the Ubuntu 22.04 testing environment.

**Comment by williamstein** [comment]
Here's a GPT-4 short paragraph summary of what's new in Sage-9.8, based on the [release tour](https://github.com/sagemath/sage/wiki/Sage-9.8-Release-Tour):  

> "Sage 9.8 was released on February 11, 2023, with contributions from 80 people. This version brings significant changes and improvements in linear algebra, commutative algebra, algebra, polyhedral geometry, number theory, elliptic curves, and manifolds. New features include submodules of tensor modules with prescribed symmetries, free module isomorphisms, solving polynomial systems using the msolve library, lazy power series ring enhancements, free resolution of modules over polynomial rings, and more. The release also includes numerous package upgrades, such as Python 3.11, PARI 2.15.2, NumPy 1.23.5, SciPy 1.9.3, SymPy 1.11.1, and other packages. Additionally, Sage development has moved from Trac to GitHub, streamlining the development process and preserving existing discussions through a custom migration script."
